django_cloud_tasks/tasks/task.py

```diff
@@ -319,18 +319,23 @@ def discard(cls, task_id: str | None = None, min_retries: int = 0):
         else:
             task_objects = client.list_tasks(queue_name=cls.queue())
 
-        outputs = []
-        for task_obj in task_objects:
+        def process_task(task_obj, min_retries, cls, client):
+            outputs = []
             task_name = task_obj.http_request.url.rsplit("/", 1)[-1]
-            if task_name != cls.name():
-                continue
-
-            if task_obj.dispatch_count < min_retries:
-                continue
+            if task_name == cls.name() and task_obj.dispatch_count >= min_retries:
```

Comment:
This check could be done when generating the jobs (and not by the processor itself) to simplify the processing code.